code/game/machinery/fire_alarm.dm

```diff
@@ -36,31 +36,75 @@ FIRE ALARM
 /obj/machinery/firealarm/update_icon()
 	cut_overlays()
 
+	pixel_x = 0
+	pixel_y = 0
+	if(dir == NORTH)
+		pixel_y = -21
+	else if(dir == SOUTH)
+		pixel_y = 21
+	else if(dir == WEST)
+		pixel_x = -21
+	else if(dir == EAST)
+		pixel_x = 21
+
+	add_overlay(image(icon, "casing"))
```

Comment:
adding a state overlay of the current icon doesn't need an image, just use a string e.g. add_overlay("casing")